app/Models/User.php

```diff
@@ -340,6 +343,16 @@ public function careTeams(): HasMany
         return $this->hasMany(CareTeam::class);
     }
 
+    public function getCareTeamRoleFor(string $educatableId): ?CareTeamRole
+    {
+        /**
+         * @var CareTeam $careTeam
+         */
+        $careTeam = $this->careTeams->where('educatable_id', $educatableId)->first();
+
+        return CareTeamRole::where('id', $careTeam->care_team_role_id)->first();
```

Comment:
I think you may be able to reduce this down to one query by applying a `with` to eager load the `careteamRole` relationship and return it directly.